code/mission/missionmessage.cpp

```diff
@@ -196,52 +195,31 @@ int comm_between_player_and_ship(int other_shipnum, bool for_death_scream);
 
 // following functions to parse messages.tbl -- code pretty much ripped from weapon/ship table parsing code
 
-static void persona_parse_close()
-{
-	if (Personas != NULL) {
-		vm_free(Personas);
-		Personas = NULL;
-	}
-}
-
 // functions to deal with parsing personas.  Personas are just a list of names that give someone
 // sending a message an identity which spans the life of the mission
 void persona_parse()
 {
-	int i;
-	char type[NAME_LENGTH];
-
-	static bool done_at_exit = false;
-	if ( !done_at_exit ) {
-		atexit( persona_parse_close );
-		done_at_exit = true;
-	}
-
-	// this way should cause the least amount of problems on the various platforms - taylor
-	Personas = (Persona*)vm_realloc( Personas, sizeof(Persona) * (Num_personas + 1) );
-
-	if (Personas == NULL)
-		Error(LOCATION, "Not enough memory to allocate Personas!" );
-
-	memset(&Personas[Num_personas], 0, sizeof(Persona));
+	Persona this_persona;
 
 	required_string("$Persona:");
-	stuff_string(Personas[Num_personas].name, F_NAME, NAME_LENGTH);
+	stuff_string(this_persona.name, F_NAME, NAME_LENGTH);
+	
+	bool dup = false;
+	if (message_persona_name_lookup(this_persona.name) >= 0) {
+		Warning(LOCATION, "Duplicate Persona %s found!", this_persona.name);
```

Comment:
Should probably say that this persona is going to be ignored, too.

Reply:
Good call. Done.